cscs-checks/prgenv/affinity_check.py

```diff
@@ -0,0 +1,155 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class AffinityTestBase(rfm.RegressionTest):
+    def __init__(self, variant):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'daint:mc',
+                              'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.build_system = 'Make'
+        self.build_system.options = ['--directory=affinity', 'MPI=1']
+        self.prebuild_cmd = ['git clone https://github.com/vkarak/affinity']
+        self.executable = 'affinity/affinity'
+        self.variant = variant
+
+        self.maintainers = ['RS', 'VK']
+        self.tags = {'production', 'scs'}
+
+    def setup(self, partition, environ, **job_opts):
+
+        def parse_affinity(x):
+            x_int = [int(xi) for xi in x.split()]
+            return sorted(x_int)
+
+        re_aff_cores = r'CPU affinity: \[\s+(?P<cpu>[\d+\s+]+)\]'
```

Comment:
I know that you wanted to save some space and be more compact, but I could not understand the name of the variable from the beginning. I would vote for `core_affinity_regex` as variable name. Am I being too picky?

The same applies to the other cases.

Reply:
@victorusu I didn't have problem understanding the names. To my opinion, what is missing is a bit of explanation about the logic as I pointed out in my review.